runtime/gc_glue_java/PointerArrayObjectScanner.hpp

```diff
@@ -58,7 +58,7 @@ class GC_PointerArrayObjectScanner : public GC_IndexableObjectScanner
 	 * @param[in] flags Scanning context flags
 	 */
 	MMINLINE GC_PointerArrayObjectScanner(MM_EnvironmentBase *env, omrobjectptr_t arrayPtr, fomrobject_t *basePtr, fomrobject_t *limitPtr, fomrobject_t *scanPtr, fomrobject_t *endPtr, uintptr_t flags)
-		: GC_IndexableObjectScanner(env, arrayPtr, basePtr, limitPtr, scanPtr, endPtr, ((endPtr - scanPtr) < _bitsPerScanMap) ? (((uintptr_t)1 << (endPtr - scanPtr)) - 1) : UDATA_MAX, sizeof(fomrobject_t), flags)
+		: GC_IndexableObjectScanner(env, arrayPtr, basePtr, limitPtr, scanPtr, endPtr, (GC_SlotObject::subtractSlotAddresses(endPtr, scanPtr, env->compressObjectReferences()) < _bitsPerScanMap) ? ((uintptr_t)1 << GC_SlotObject::subtractSlotAddresses(endPtr, scanPtr, env->compressObjectReferences())) - 1 : UDATA_MAX, env->compressObjectReferences() ? sizeof(uint32_t) : sizeof(uintptr_t), flags)
```

Comment:
This line is too long now... Can it be split somehow to be more read friendly?

Reply:
Will do.

Reply:
See if it's better now...